near/omni-bridge/src/lib.rs

```diff
@@ -1005,16 +1128,39 @@ impl Contract {
         promise
     }
 
-    fn process_fin_transfer_to_other_cahin(
+    fn process_fin_transfer_to_other_chain(
         &mut self,
         predecessor_account_id: AccountId,
         transfer_message: TransferMessage,
     ) {
         let mut required_balance = self.add_fin_transfer(&transfer_message.get_transfer_id());
+        let token = self.get_token_id(&transfer_message.token);
 
-        required_balance = self
-            .add_transfer_message(transfer_message.clone(), predecessor_account_id.clone())
-            .saturating_add(required_balance);
+        let fast_transfer = FastTransfer::from_transfer(transfer_message.clone(), token.clone());
+        let recipient = match self.fast_transfers.get(&fast_transfer.hash()) {
+            Some(relayer) => {
+                require!(
+                    predecessor_account_id == *relayer,
+                    "ERR_FAST_TRANSFER_PERFORMED_BY_ANOTHER_RELAYER"
+                );
+                Some(relayer)
+            }
+            None => None,
+        };
+
+        // If fast transfer happened, send tokens to the relayer that executed fast transfer
+        if let Some(relayer) = recipient {
+            self.send_tokens(
+                token,
+                relayer,
+                U128(transfer_message.amount.0 - transfer_message.fee.fee.0),
+                String::new(),
+            );
+        } else {
+            required_balance = self
```

Comment:
What if the `fast_transfer` data isn't correct, for example wrong amount or recipient, the `add_transfer_message` will fail in this case. So it should be possible to overwrite it.

Reply:
`fast_transfer` is derived from the actual transfer message in this case, unless I'm missing something?

Reply:
I mean that the transfer is already inserted fast fin transfer to other chains by transfer Id, so this insert will fail if the hash of the messages are different.

Reply:
fa617c48582d9f3f3d111c9b8aca1fe53066445c

Reply:
This is now fixed because we generated new transfer_id for fast transfer to 3rd chain